hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileWriter.java

```diff
@@ -247,7 +222,10 @@ public void appendMetadata(final long maxSequenceId, final boolean majorCompacti
    * @throws IOException problem writing to FS
    */
   public void appendMobMetadata(SetMultimap<TableName, String> mobRefSet) throws IOException {
-    writer.appendFileInfo(MOB_FILE_REFS, MobUtils.serializeMobFileRefs(mobRefSet));
+    liveFileWriter.appendMobMetadata(mobRefSet);
```

Comment:
Ditto.

Reply:
By default all files are live file when the dual file writing is not enabled. Dual file writing is not supported for mob in this PR.

Reply:
Is there a way to prevent users enable dual file writing and mob together?

Reply:
Since we moved dual file writing into StoreFileWriter and all compaction methods use StoreFileWriter, there is no way to prevent it. Please note by default dual file writing is disabled, one needs to enable it to use it by setting  hbase.enable.historical.compaction.files to true.

Reply:
Could you check if mob is enabled and, if so, log a warn and ignore the dual writer config value?

Reply:
I have done that the last commit.